client/cli/user/user.go

```
@@ -37,11 +44,96 @@ func init() {
+					Subcommands: []*cli.Command{
+						{
+							Name:   "password",
+							Usage:  "Set password",
+							Action: changePassword,
+							Flags: []cli.Flag{
+								&cli.StringFlag{
+									Name:  "old-password",
+									Usage: "Existing password, the one that is used currently.",
+								},
+								&cli.StringFlag{
+									Name:  "new-password",
+									Usage: "New password you want to set.",
+								},
+							},
+						},
+					},
+				},
 			},
 		},
 	)
 }
 
+// get current user settings
+func changePassword(ctx *cli.Context) error {
+	env := util.GetEnv(ctx)
+
+	token, err := config.Get("micro", "auth", env.Name, "token")
+	if err != nil {
+		return err
+	}
+
+	oldPassword := ctx.String("old-password")
+	newPassword := ctx.String("new-password")
+
+	if len(oldPassword) == 0 {
+		fmt.Print("Enter current password: ")
+		bytePw, _ := terminal.ReadPassword(int(syscall.Stdin))
+		pw := string(bytePw)
+		pw = strings.TrimSpace(pw)
+		fmt.Println()
+		oldPassword = pw
+	}
+
+	if len(newPassword) == 0 {
+		for {
+			fmt.Print("Enter a new password: ")
+			bytePw, _ := terminal.ReadPassword(int(syscall.Stdin))
+			pw := string(bytePw)
+			pw = strings.TrimSpace(pw)
+			fmt.Println()
+
+			fmt.Print("Verify your password: ")
+			bytePwVer, _ := terminal.ReadPassword(int(syscall.Stdin))
+			pwVer := string(bytePwVer)
+			pwVer = strings.TrimSpace(pwVer)
+			fmt.Println()
+
+			if pw != pwVer {
+				fmt.Println("Passwords do not match. Please try again.")
+				continue
+			}
+			newPassword = pw
+			break
+		}
+	}
+
+	// Inspect the token
+	acc, err := auth.Inspect(token)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	accountService := pb.NewAccountsService("auth", client.DefaultClient)
+	_, err = accountService.ChangePassword(context.TODO(), &pb.ChangePasswordRequest{
+		Id:        acc.ID,
+		OldSecret: oldPassword,
+		NewSecret: newPassword,
+	}, goclient.WithAuthToken())
+	if err != nil {
+		fmt.Println(err)
```

Rather than printing the error and exiting, you can just return the error.